# ChatGPTが生成したサンプルコードを実際に試す →失敗

In [1]:
!pip install whisper

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.8/42.8 KB 1.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for whisper: filename=whisper-1.1.10-py3-none-any.whl size=41134 sha256=7996660022e7543385437e1b028e194044071291f78f5810133b5b8a3b25194f
  Stored in directory: /root/.cache/pip/wheels/f1/01/8c/1b449563518aa0f8813d044f7a910884e3b8313711becb66d6
Successfully built whisper


In [3]:
# 確認用
!ls

sample_data  test-data1.m4a  test-data1.wav


In [12]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
transcript = whisper.transcribe(audio_file, model='large', lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

AttributeError: ignored

In [13]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
transcript = whisper.recognize_google(audio_file, model='large', lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

AttributeError: ignored

In [14]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
transcript = whisper.recognize(audio_file, model='google', lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

AttributeError: ignored

In [15]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
transcript = whisper.transcribe(audio_file, model='google', lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

AttributeError: ignored

# ChatGPTが追加で回答したSpeechRecognitionの利用へ変更 →成功
ChatGPTが方針転換して追加で回答したSpeechRecognitionの利用に従ってみる。

In [5]:
!pip install SpeechRecognition

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 32.8/32.8 MB 25.8 MB/s eta 0:00:00


In [17]:
import speech_recognition as sr

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# 音声認識器を初期化する
r = sr.Recognizer()

# 音声ファイルからオーディオを読み込む
with sr.AudioFile(audio_file) as source:
    audio = r.record(source)  # 全部読み取る

# 音声を文字列に変換する
text = r.recognize_google(audio, language='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(text))

Transcript: これは蒸しパン ライバルのテストです 昨日 豊富なもの の欠点 複数の欠点もある ovi はちゃんと GPT が時によってはもっともらしく見えると謝っている 解答作成することを認めているチャット GPT の報酬も電波人間による短所 中心としているため 最適化 再設定 パフォーマンスに影響を及ぼしてしまうグッド ハートの法則 それに加えチャット GPT は 2011年以降に発生したということについては 知識が蓄えられて終わらせする の著名人については 知識が全くないこともある


エラーなく実行可能。結果は: 誤りが多い印象。

Transcript: これは蒸しパン ライバルのテストです 昨日 豊富なもの 複数の欠点もある ovi はちゃんと GPT が時によってはもっともらしく見えると謝っている 解答作成することを認めているチャット GPT の報酬も電波人間による短所 中心としているため 最適化 再設定 パフォーマンスに影響を及ぼしてしまうグッド ハートの法則 それに加えチャット GPT は 2011年以降に発生したということについては 知識が蓄えられて終わらせする の著名人については 知識が全くないこともある

In [18]:
import speech_recognition as sr

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# 音声認識器を初期化する
r = sr.Recognizer()

# 音声ファイルからオーディオを読み込む
with sr.AudioFile(audio_file) as source:
    audio = r.record(source)  # 全部読み取る

# 音声を文字列に変換する
text = r.recognize_bing(audio, language='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(text))

TypeError: ignored

ChatGPTの回答に従い、recognize_bingを試すと、API keyが必要となる模様。

In [19]:
import speech_recognition as sr

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# 音声認識器を初期化する
r = sr.Recognizer()

# 音声ファイルからオーディオを読み込む
with sr.AudioFile(audio_file) as source:
    audio = r.record(source)  # 全部読み取る

# 音声を文字列に変換する
text = r.recognize_sphinx(audio, language='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(text))

RequestError: ignored

ChatGPTの回答に従い、recognize_sphinxを試すと、追加のモジュールが必要となる模様。

# 通常通り手動でWeb検索を用いて調べて試す →成功

https://happy-shibusawake.com/openai_whisper/696/

> Whisperと関連ライブラリのインストール
Whisperと関連ライブラリをpipコマンドでインストールします。足りないライブラリとWhisperをインストールしてくれます。
pip install git+https://github.com/openai/whisper.git 
Installing collected packages: more-itertools, ffmpeg-python, whisper
Successfully installed ffmpeg-python-0.2.0 more-itertools-8.14.0 whisper-1.0
これでインストール完了です。

https://www.kkaneko.jp/ai/win/whisper.html

> ダウンロードとインストール
python -m pip install -U git+https://github.com/openai/whisper.git
関連ファイルをダウンロード
cd %HOMEPATH%
rmdir /s /q whisper
git clone --recursive https://github.com/openai/whisper.git

https://qiita.com/miyanaga/items/d785f8c613da9daf1d76

> Whisperで文字起こし
このようなPythonスクリプトで文字起こししました。こんな短いプログラムでできるなんてすごい！
import whisper
model = whisper.load_model("large")
result = model.transcribe("./1st.mov", verbose=True, language='ja')
text = result['text']
print(text)

In [29]:
!pip install git+https://github.com/openai/whisper.git

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Cloning https://github.com/openai/whisper.git to /tmp/pip-req-build-ne4dafhh
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git /tmp/pip-req-build-ne4dafhh
  Resolved https://github.com/openai/whisper.git to commit 6dea21fd7f7253bfe450f1e2512a0fe47ee2d258
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.3/63.3 MB 9.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 65.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.3/132.3 KB 12.7 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for openai-whisper: filename=openai_whisper-20230314-py3-none-any.whl size=796926 sha256=41c5cdc72edb540a10272a03e838fe6e914b4c0c9d02d4894dc1cca091348af5
  Stored in dir

In [1]:
import whisper

model = whisper.load_model("large")
result = model.transcribe("test-data1.wav", verbose=True, language='ja')
text = result['text']
print(text)

100%|██████████████████████████████████████| 2.87G/2.87G [00:29<00:00, 104MiB/s]
/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")


[00:00.000 --> 00:08.040] これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点
[00:08.040 --> 00:12.960] もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている
[00:12.960 --> 00:17.400] 回答を作成することを認めている ChatGPTの方針モデルは人間による
[00:17.400 --> 00:22.600] 監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまう
[00:22.600 --> 00:27.200] グッドハートの法則 それに加え ChatGPTは2011年以降に発生した
[00:27.200 --> 00:31.320] 出来事については知識が蓄えられて 終わらず 一部の著名人については
[00:31.320 --> 00:32.880] 知識が全くないこともある
これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている回答を作成することを認めている ChatGPTの方針モデルは人間による監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまうグッドハートの法則 それに加え ChatGPTは2011年以降に発生した出来事については知識が蓄えられて 終わらず 一部の著名人については知識が全くないこともある


エラーなく実行可能。結果は: かなり良好。

これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている回答を作成することを認めている ChatGPTの方針モデルは人間による監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまうグッドハートの法則 それに加え ChatGPTは2011年以降に発生した出来事については知識が蓄えられて 終わらず 一部の著名人については知識が全くないこともある

# これを踏まえて再度、ChatGPTが生成したサンプルコードを修正しながら試す →成功

ChatGPTの回答は、まずインストールするライブラリが間違っており、実行が出来なかった模様。→修正してインストール(上記で済み)

In [2]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
transcript = whisper.transcribe(audio_file, model='large', lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

TypeError: ignored

ChatGPTの回答は、whisper.transcribeは、引数にmodel='large'を取らない模様。→修正

In [1]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
model = whisper.load_model("large")
transcript = model.transcribe(audio_file, lang='ja-JP')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")


TypeError: ignored

ChatGPTの回答は、whisper.transcribeは、引数にlang='ja-JP'を取らない模様。→修正

In [2]:
import whisper

# 音声ファイルを読み込む
audio_file = 'test-data1.wav'

# Whisperを使用して音声ファイルを文字起こしする
model = whisper.load_model("large")
transcript = model.transcribe(audio_file, language='ja')

# 認識されたテキストを表示する
print('Transcript: {}'.format(transcript))

/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")


Transcript: {'text': 'これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている回答を作成することを認めている ChatGPTの方針モデルは人間による監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまうグッドハートの法則 それに加え ChatGPTは2011年以降に発生した出来事については知識が蓄えられて 終わらず 一部の著名人については知識が全くないこともある', 'segments': [{'id': 0, 'seek': 0, 'start': 0.0, 'end': 8.040000000000001, 'text': 'これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点', 'tokens': [50364, 25212, 2471, 271, 610, 12806, 2972, 22985, 40498, 4767, 220, 17543, 8225, 3065, 19517, 232, 47564, 3203, 44726, 2972, 220, 164, 97, 229, 33188, 2972, 5988, 254, 12579, 50766], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 1, 'seek': 0, 'start': 8.040000000000001, 'end': 12.96, 'text': 'もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている', 'tokens': [50766, 4801, 24719, 7238, 48698, 3065, 41683, 38, 47, 51, 5142, 6611, 4108, 5591, 6102, 20785, 8661, 4801, 5154, 26568, 10470, 44547, 5142, 3549, 97, 6102,

エラーなく実行可能となる。結果は: 前回と同様。かなり良好。

Transcript: {'text': 'これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている回答を作成することを認めている ChatGPTの方針モデルは人間による監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまうグッドハートの法則 それに加え ChatGPTは2011年以降に発生した出来事については知識が蓄えられて 終わらず 一部の著名人については知識が全くないこともある', 'segments': [{'id': 0, 'seek': 0, 'start': 0.0, 'end': 8.040000000000001, 'text': 'これはWhisper Libraryのテストです 機能は豊富なものの 複数の欠点', 'tokens': [50364, 25212, 2471, 271, 610, 12806, 2972, 22985, 40498, 4767, 220, 17543, 8225, 3065, 19517, 232, 47564, 3203, 44726, 2972, 220, 164, 97, 229, 33188, 2972, 5988, 254, 12579, 50766], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 1, 'seek': 0, 'start': 8.040000000000001, 'end': 12.96, 'text': 'もある OpenAIはChatGPTが時によって は最もらしく見えるが誤っている', 'tokens': [50766, 4801, 24719, 7238, 48698, 3065, 41683, 38, 47, 51, 5142, 6611, 4108, 5591, 6102, 20785, 8661, 4801, 5154, 26568, 10470, 44547, 5142, 3549, 97, 6102, 22979, 51012], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 2, 'seek': 0, 'start': 12.96, 'end': 17.400000000000002, 'text': '回答を作成することを認めている ChatGPTの方針モデルは人間による', 'tokens': [51012, 8350, 28223, 5998, 11914, 11336, 22570, 13235, 5998, 22041, 38975, 22979, 27503, 38, 47, 51, 2972, 9249, 5873, 251, 29183, 31327, 9405, 3065, 4035, 11016, 4108, 5591, 4895, 51234], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 3, 'seek': 0, 'start': 17.400000000000002, 'end': 22.6, 'text': '監視を中心としているため 最低 化されすぎてパフォーマンスに影響を及ぼしてしまう', 'tokens': [51234, 5419, 96, 27333, 5998, 5975, 7945, 3193, 8822, 22979, 49983, 220, 8661, 41377, 220, 23756, 6722, 4132, 2659, 44601, 2996, 23268, 17320, 824, 102, 3384, 13258, 4824, 9550, 4108, 16820, 13665, 123, 5998, 25703, 487, 120, 8822, 45349, 2646, 51494], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 4, 'seek': 0, 'start': 22.6, 'end': 27.2, 'text': 'グッドハートの法則 それに加え ChatGPTは2011年以降に発生した', 'tokens': [51494, 23839, 8276, 11195, 15927, 38551, 2972, 11148, 46225, 47765, 4108, 9990, 6474, 27503, 38, 47, 51, 3065, 2009, 5348, 5157, 3588, 47421, 4108, 44756, 8244, 8533, 51724], 'temperature': 0.0, 'avg_logprob': -0.17137325934644015, 'compression_ratio': 1.3304597701149425, 'no_speech_prob': 0.03912182152271271}, {'id': 5, 'seek': 2720, 'start': 27.2, 'end': 31.32, 'text': '出来事については知識が蓄えられて 終わらず 一部の著名人については', 'tokens': [50364, 44561, 6973, 4108, 9335, 18549, 3065, 6498, 43143, 5142, 39198, 226, 6474, 5154, 26638, 220, 29371, 9206, 5154, 18216, 26923, 13470, 2972, 19382, 15940, 4035, 4108, 9335, 18549, 3065, 50570], 'temperature': 0.0, 'avg_logprob': -0.2219417039738145, 'compression_ratio': 1.190909090909091, 'no_speech_prob': 0.914054811000824}, {'id': 6, 'seek': 2720, 'start': 31.32, 'end': 32.879999999999995, 'text': '知識が全くないこともある', 'tokens': [50570, 6498, 43143, 5142, 11319, 6134, 9311, 13235, 4801, 24719, 50648], 'temperature': 0.0, 'avg_logprob': -0.2219417039738145, 'compression_ratio': 1.190909090909091, 'no_speech_prob': 0.914054811000824}], 'language': 'ja'}

# 前処理: .wavファイルへ変換

In [7]:
!pip install ffmpeg-python

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [8]:
import ffmpeg

stream = ffmpeg.input("test-data1.m4a")
stream = ffmpeg.output(stream, "test-data1.wav")
ffmpeg.run(stream)

(None, None)

In [9]:
!ls

sample_data  test-data1.m4a  test-data1.wav


.wavへ変換成功